### PR TITLE
Added support for Middlesbrough Council - UK

### PR DESCRIPTION
**Pull request by ximon**
Added source and documentation for Middlesbrough council - UK
Currently uses the UPRN.

### CODE REVIEW

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/middlesbrough_gov_uk.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        #now query using the uprn
+        payload = {
+            "formValues": { "Find My Collection Dates": { "uprn_search": { "value": self._uprn } } }
+        }
+        scheduleRequest = s.post(API_URLS['schedule'] + '&_' + str(now) + '&sid=' + sessionKey , headers=HEADERS, json=payload)
+        #print(scheduleRequest.json())
+        data = scheduleRequest.json()['integration']['transformed']['rows_data']['0']
+
+        print(data)
```
Comment (mampfes) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/middlesbrough_gov_uk.py:77]: Remove debug output

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/middlesbrough_gov_uk.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Source:
+    def __init__(self, uprn: str = None):
+        self._uprn = str(uprn)
+        if not self._uprn:
```
Comment (mampfes) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/middlesbrough_gov_uk.py:50]: If you remove `= None` from the uprn argument then you can also discard this if statement.

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/middlesbrough_gov_uk.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        greenDates = data['Green'].split('<br />')
+
+        entries = []
+
+        for date in refuseDates:
+            if len(date) > 0:
+                entries.append(Collection(
+                    date = datetime.strptime(date, '%d/%m/%Y').date(),
+                    t = 'refuse bin',
+                    icon = ICONS.get('REFUSE')
```
Comment (mampfes) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/middlesbrough_gov_uk.py:90]: Just wondering why you use a `ICONS` dict here. Wouldn't it be simpler to just name the icon here directly (same as for `t`)?

**Review by mampfes — CHANGES_REQUESTED** [review]

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/middlesbrough_gov_uk.py (hunk 3 above)
Comment (ximon) [reply custom_components/waste_collection_schedule/waste_collection_schedule/source/middlesbrough_gov_uk.py:90]: Very true, i was just going with the style of the source i based this upon.

**Comment by ximon** [comment]
I've made the suggested changes and tested 👍

**Comment by mampfes** [comment]
Thanks for your contribution.

**Pull request merged**